app/qan/mysql.go

```diff
@@ -256,55 +262,65 @@ func (h *MySQLMetricWriter) getClassId(checksum string) (uint, error) {
 	return classId, nil
 }
 
-func (h *MySQLMetricWriter) newClass(instanceId uint, class *event.Class, lastSeen string) (uint, error) {
-	// In theory the shortest valid SQL statment should be at least 2 chars
-	// (I'm not aware of any 1 char statments), so if the fingerprint is shorter
-	// than 2 then it's invalid, effectively empty, probably due to a slow log
-	// parser bug.
-	if len(class.Fingerprint) < 2 {
-		return 0, fmt.Errorf("empty fingerprint")
-	} else if class.Fingerprint[len(class.Fingerprint)-1] == '\n' {
-		// https://jira.percona.com/browse/PCT-826
-		class.Fingerprint = strings.TrimSuffix(class.Fingerprint, "\n")
-		log.Printf("WARNING: fingerprint had newline: %s: instance_id=%d", class.Fingerprint, instanceId)
-	}
+func (h *MySQLMetricWriter) newClass(instanceId uint, subsystem string, class *event.Class, lastSeen string) (uint, error) {
+	var queryAbstract, queryQuery string
+	var tables interface{}
 
-	// Distill the query (select c from t where id=? -> SELECT t) and extract
```

Comment:
Here is the actual change. Simple switch to split the code execution.
If subsystem is `mysql` then do original code for parsing query.
If subsystem is `mongo` then simply use fingerprint for query and abstract.